Userland/Applications/DisplaySettings/MonitorWidget.cpp

```diff
@@ -37,7 +37,7 @@ bool MonitorWidget::set_wallpaper(DeprecatedString path)
             return Gfx::Bitmap::load_from_file(path);
         },
 
-        [this, path](ErrorOr<NonnullRefPtr<Gfx::Bitmap>> bitmap_or_error) -> ErrorOr<void> {
+        [this, path](ErrorOr<NonnullRefPtr<Gfx::Bitmap>>& bitmap_or_error) -> ErrorOr<void> {
```

Comment:
What's this change? Why are we passing an ErrorOr<> by reference?

Reply:
That's the on-error callback signature, since the error is stored with the promise.

Reply:
This feels odd. This function should be a `BackgroundAction<NNRP<Bitmap>`, and any errors from the action should be passed onto the Error callback, not the completion callback. I think this was simply missed in <https://github.com/SerenityOS/serenity/commit/2693745336ddc6c7a790ecf8a760aa74651ac055>